decidim-blogs/app/permissions/decidim/blog/permissions.rb

```diff
@@ -7,15 +7,90 @@ def permissions
         return permission_action unless permission_action.subject == :blogpost || permission_action.subject == :post
 
         if permission_action.scope == :public
+          if permission_action.action.in?([:update, :destroy])
+            toggle_allow(can_manage_post)
+            return permission_action
+          end
+
+          if permission_action.action == :create
+            toggle_allow(can_create_post)
+            return permission_action
+          end
+
           allow!
           return permission_action
         end
 
-        return permission_action if permission_action.scope != :admin
+        return permission_action unless permission_action.scope == :admin
+
+        if permission_action.action.in?([:update, :destroy])
+          toggle_allow(admin_can_manage_post)
+          return permission_action
+        end
 
         allow!
         permission_action
       end
+
+      def post
+        @post ||= context.fetch(:blogpost, nil)
+      end
+
+      def current_component
+        @current_component ||= context.fetch(:current_component, nil)
+      end
+
+      def can_create_post
+        current_component&.participatory_space.is_a?(Decidim::Initiative) &&
```

Comment:
The initiatives module may not be installed in a particular application which can lead to errors. In cases where a module is mentioned from another try to add something like a guard clause checking the presence of the module with `Decidim.module_installed?("initiatives")` or define a method that only initiatives participatory spaces of type initiative can respond positively asking directly to the participatory space, regardless of its class